### PR TITLE
Rework how visual diffs are run, allowing the user to specify diff paths

### PR DESCRIPTION
**Pull request by malcolmbaig**
The existing visual differ worked by fetching a bunch of urls from govuk, generated a wraith config with these paths, and passed this into wraith. Configuring the visual differ to accept a user-specified list of paths was non-trivial.

This PR adds a rake task that accepts a URI for a yaml file containing the paths to run a visual diff against. It includes some cleanup of unused/unnecessary code - further cleanup will follow in subsequent PRs.

Trello: https://trello.com/c/J6Z8RxEt

### CODE REVIEW

File: lib/govuk/diff/pages/visual_diff/runner.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          def initialize(list_of_pages_uri:)
+            @list_of_pages_uri = list_of_pages_uri
+          end
+
+          def run
+            pages = YAML.load open(@list_of_pages_uri)
+            wraith_config = WraithConfig.new(pages: pages)
+            wraith_config.write
+
+            Kernel.puts "---> Creating Visual Diffs"
```
Comment (tijmenb) [inline lib/govuk/diff/pages/visual_diff/runner.rb:19]: Why the `Kernel`?

File: lib/govuk/diff/pages/tasks/rakefile.rake [hunk 2]
```diff
@@ -3,10 +3,8 @@ require 'govuk/diff/pages'
 namespace :diff do
   desc 'produce visual diffs'
```
Comment (tijmenb) [inline lib/govuk/diff/pages/tasks/rakefile.rake]: Perhaps update the description to mention the `URI` config.

File: lib/govuk/diff/pages/visual_diff/runner.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require 'yaml'
+require 'tempfile'
+require 'open-uri'
+
+module Govuk
+  module Diff
+    module Pages
+      module VisualDiff
+        class Runner
+          def initialize(list_of_pages_uri:)
```
Comment (tijmenb) [inline lib/govuk/diff/pages/visual_diff/runner.rb:10]: I'd do the YAML parsing in the rake task and provide an array of URIs to this class.

**Comment by tijmenb** [comment]
Can we add a full example to the readme?

File: lib/govuk/diff/pages/visual_diff/runner.rb (hunk 1 above)
Comment (malcolmbaig) [inline lib/govuk/diff/pages/visual_diff/runner.rb:19]: Makes stubbing writes to stdout much simpler in specs.

File: lib/govuk/diff/pages/tasks/rakefile.rake (hunk 2 above)
Comment (malcolmbaig) [inline lib/govuk/diff/pages/tasks/rakefile.rake]: done

**Comment by malcolmbaig** [comment]
@tijmenb we have a full README overhaul next in our tasklist, but yeah - can provide an update in this PR too.

File: lib/govuk/diff/pages/visual_diff/runner.rb (hunk 3 above)
Comment (malcolmbaig) [inline lib/govuk/diff/pages/visual_diff/runner.rb:10]: Subsequent PR! Promise.

Comment (MatMoore) [inline lib/govuk/diff/pages/visual_diff/runner.rb:10]: That's a lot of nested modules we have there! I know it's not new, but could we combine some of these in a future PR?

Eg

``` ruby
module GovukDiffPages
  module VisualDiff
    class Runner
    end
  end
end
```

**Pull request merged**